### PR TITLE
cpu, cc2538: gpio cleanup

### LINKED ISSUE #6650: GPIO Port shift in cc2538 is unconsistent
Hi,
I just realized when trying to initialize a GPIO Pin with an interrupt through `gpio_init_int(...)` the pin enumaration is inconsistent to the defines in `cc2538_gpio.h`. While the port shift in `cc2538_gpio.h` is 3, it is 12 in `periph/gpio.c`. So when trying to initialize the GPIO `GPIO_PB0` for instance, which is the gpio with port=1 and pin=0 (so encoding by `cc2538_gpio.h`means its pin number 8) `gpio_init_int(...)` will try to access `isr_ctx[][]` at 4294967295, as `port_num(...)` also subracts 1. This leads to a hardfault on my re-mote platform.

### PR DESCRIPTION
**Pull request by smlng**
remove any (okay: most) dependencies on the vendor specific (TI) gpio api, to fully embrace RIOTs own hardware independent periph gpio API.

Depends on #7310, #7313, #7316, and #7318 will resolve #6650 and makes workaround in #6773 obsolete as soon as all PRs merged.

### CODE REVIEW

File: cpu/cc2538/periph/gpio.c [hunk 1]
```diff
@@ -26,37 +26,8 @@
...
-}
-
-static inline int pp_num(gpio_t pin)
-{
-    return (port_num(pin) * 8) + pin_num(pin);
-}
-
 static gpio_isr_ctx_t isr_ctx[4][8];
 
 int gpio_init(gpio_t pin, gpio_mode_t mode)
```
Comment (dnahm) [inline cpu/cc2538/periph/gpio.c]: The brace belongs behind the if-statement.

**Comment by smlng** [comment]
will rebase and resolve conflicts as soon as all required PRs are merged.

**Comment by tcschmidt** [comment]
@A-Paul thinking about starting the review?

File: cpu/cc2538/periph/i2c.c [hunk 2]
```diff
@@ -26,33 +27,34 @@
...
+#include "thread.h"
 #include "timex.h" /* for US_PER_SEC */
+#endif
 
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 
 /* guard this file in case no I2C device is defined */
-#if I2C_NUMOF
+#ifdef I2C_NUMOF
```
[the remark this answers is not in the document]
Comment (smlng) [reply cpu/cc2538/periph/i2c.c]: no but this is changed in #9347 anyways, essentially this PR might be obsolete or at least needs major updates after the I2C rework is completed. Hence, no need for reviewing this one right now.

[the remark this answers is not in the document]
Comment (tcschmidt) [reply cpu/cc2538/periph/i2c.c]: Then close?

[the remark this answers is not in the document]
Comment (smlng) [reply cpu/cc2538/periph/i2c.c]: ack

**Comment by smlng** [comment]
close, will rework and open new PR later

**Pull request closed without merging**